src/extension.ts

```diff
@@ -902,41 +902,50 @@ function promptKindName(resourceKinds: kuberesources.ResourceKind[], description
 }
 
 function quickPickKindName(resourceKinds: kuberesources.ResourceKind[], opts, handler) {
+    if (resourceKinds.length === 1) {
+        quickPickKindNameFromKind(resourceKinds[0], opts, handler);
+        return;
+    }
+
     vscode.window.showQuickPick(resourceKinds).then((resourceKind) => {
         if (resourceKind) {
-            let kind = resourceKind.abbreviation;
-            kubectl.invoke("get " + kind, (code, stdout, stderr) => {
-                if (code === 0) {
-                    let names = parseNamesFromKubectlLines(stdout);
-                    if (names.length > 0) {
-                        if (opts && opts.nameOptional) {
-                            names.push('(all)');
-                            vscode.window.showQuickPick(names).then((name) => {
-                                if (name) {
-                                    let kindName;
-                                    if (name === '(all)') {
-                                        kindName = kind;
-                                    } else {
-                                        kindName = kind + '/' + name;
-                                    }
-                                    handler(kindName);
-                                }
-                            });
-                        } else {
-                            vscode.window.showQuickPick(names).then((name) => {
-                                if (name) {
-                                    let kindName = kind + '/' + name;
-                                    handler(kindName);
-                                }
-                            });
+            quickPickKindNameFromKind(resourceKind, opts, handler);
+        }
+    });
+}
+
+function quickPickKindNameFromKind(resourceKind: kuberesources.ResourceKind, opts, handler) {
+    let kind = resourceKind.abbreviation;
+    kubectl.invoke("get " + kind, (code, stdout, stderr) => {
+        if (code === 0) {
```

Comment:
reverse this condition since the majority of work is done if the code === 0

ie: 
```
if (code !== 0) {
    vscode.window.showErrorMessage(stderr);
    return;
}

// everything else
```